components/cam/src/physics/cam/micro_p3.F90

```diff
@@ -972,38 +974,10 @@ SUBROUTINE p3_main(qc,nc,qr,nr,th_old,th,qv_old,qv,dt,qitot,qirim,nitot,birim,ss
           !   (use semi-analytic formulation)
 
           ! calculate rain evaporation including ventilation
-          if (qr_incld(i,k).ge.qsmall) then
-             call find_lookupTable_indices_3(dumii,dumjj,dum1,rdumii,rdumjj,inv_dum3,mu_r(i,k),lamr(i,k))
-             !interpolate value at mu_r
-! bug fix 12/23/18
-!             dum1 = revap_table(dumii,dumjj)+(rdumii-real(dumii))*inv_dum3*                  &
-!                    (revap_table(dumii+1,dumjj)-revap_table(dumii,dumjj))
-
-             dum1 = revap_table(dumii,dumjj)+(rdumii-real(dumii))*                            &
-                    (revap_table(dumii+1,dumjj)-revap_table(dumii,dumjj))
-
-             !interoplate value at mu_r+1
-! bug fix 12/23/18
-!             dum2 = revap_table(dumii,dumjj+1)+(rdumii-real(dumii))*inv_dum3*                &
-!                  (revap_table(dumii+1,dumjj+1)-revap_table(dumii,dumjj+1))
-             dum2 = revap_table(dumii,dumjj+1)+(rdumii-real(dumii))*                          &
-                    (revap_table(dumii+1,dumjj+1)-revap_table(dumii,dumjj+1))    
-             !final interpolation
-             dum  = dum1+(rdumjj-real(dumjj))*(dum2-dum1)
-
-             epsr = 2._rtype*pi*cdistr(i,k)*rho(i,k)*dv*(f1r*gamma(mu_r(i,k)+2._rtype)/(lamr(i,k))+f2r*   &
-                  (rho(i,k)/mu)**0.5_rtype*sc**thrd*dum)
-          else
-             epsr = 0._rtype
-          endif
-
-          if (qc_incld(i,k).ge.qsmall) then
-             epsc = 2._rtype*pi*rho(i,k)*dv*cdist(i,k)
-          else
-             epsc = 0._rtype
-          endif
-          !===
 
+          call calc_liq_relaxation_timescale(rho(i,k),f1r,f2r,     &
+          dv,mu,sc,mu_r(i,k),lamr(i,k),cdistr(i,k),cdist(i,k),qr_incld(i,k),qc_incld(i,k), &
+          epsr,epsc)  
           !PMC moved oabi outside t<273.15 loop. oabi is only *used* where t<273.15, but
           !t could be updated before oabi is used, resulting in unititialized use. Note
           !that t is not currently being updated before oabi is used, so this is just future-proofing.
```

Comment:
Can my oabi comment be deleted yet?

Reply:
I think so.  This whole calculation may not be needed when we switch to handling super-saturation using the MG code.  But now that it is brought up, do we want to use the calculation of ab directly from P3 or calculate ab on the fly using the MG routine from the utils?

Reply:
My understanding is that ab needs to be recalculated every time T changes. I hope/expect it to be computed identically in P3 versus MG. If not, we should look at how it's calculated but probably use the P3 version.

Reply:
For the most part this is true.  As we noted in our email to Hugh, we are not updating the temperature to reflect any impact from clipping at the beginning of P3.  After we hear back from him about this I suspect we will fix this potential bug and ab and abi will reflect the most up-to-date temperature.